pipeline_dsl/resources/resource.py

```diff
@@ -0,0 +1,15 @@
+from abc import ABC, abstractmethod
+
+
+class AbstractResource(ABC):
+    @abstractmethod
+    def resource_type(self) -> dict:
+        pass
+
+    @abstractmethod
+    def concourse(self, name: str) -> dict:
+        pass
+
+    @abstractmethod
+    def get(self, name: str):
+        pass
```

Comment:
`get` usually returns something. This could be modeled with a generic type. See [User-defined generic types](https://docs.python.org/3/library/typing.html#user-defined-generic-types)

```suggestion
from typing import TypeVar, Generic


T = TypeVar('T')

class AbstractResource(Generic[T]):
    @abstractmethod
    def resource_type(self) -> dict:
        pass

    @abstractmethod
    def concourse(self, name: str) -> dict:
        pass

    @abstractmethod
    def get(self, name: str) -> T:
        pass
```

Reply:
Where does the `abstractmethod` come from when using the `typing` package?
Is it already included in the `typing` package?